From c04406585217092f85f3188dab4875b7be78af1d Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 02 Dec 2021 16:56:39 +0000
Subject: [PATCH] base: enable mte via -Xclang instead

-march is fragile as it may get overridden later on by
e.g. distributors.

Bug: 1137393
Change-Id: Ia6b98491ef6f955929d0699e9f435844de48e1da
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3310965
Reviewed-by: Kentaro Hara <haraken@chromium.org>
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Cr-Commit-Position: refs/heads/main@{#947530}
---

diff --git a/base/BUILD.gn b/base/BUILD.gn
index df6293d..eaa02b7 100644
--- a/base/BUILD.gn
+++ b/base/BUILD.gn
@@ -104,7 +104,12 @@
     # but they're not backwards compatible. Use base::CPU::has_mte()
     # beforehand to confirm or use indirect functions (ifuncs) to select
     # an MTE-specific implementation at dynamic link-time.
-    cflags = [ "-march=armv8-a+memtag" ]
+    cflags = [
+      "-Xclang",
+      "-target-feature",
+      "-Xclang",
+      "+mte",
+    ]
   }
 }
 
